src/librbd/api/Trash.cc

```diff
@@ -340,7 +340,8 @@ int Trash<I>::purge(IoCtx& io_ctx, time_t expire_ts,
   trash_entries.erase(
       std::remove_if(trash_entries.begin(), trash_entries.end(),
                      [](librbd::trash_image_info_t info) {
-                       return info.source != RBD_TRASH_IMAGE_SOURCE_USER;
+                       return (info.source != RBD_TRASH_IMAGE_SOURCE_USER &&
+                               info.source != RBD_TRASH_IMAGE_SOURCE_REMOVE);
```

Comment:
Though I see how it could be useful, whouldn't it create troubles, e.g. when `purge` is executed when some image removal is in progress?

Reply:
I'm OK with racing deletions -- assuming at least it doesn't throw an error back to the user if the image was in the trash when the operation started.

Reply:
Personally I would still prefer purge  to remove only `SOURCE_USER` images, but I don't insist and will fine with the current implementation.